src/if.js

```diff
@@ -43,20 +43,42 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {      
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          delete this.__queuedChanges;       
+          return;
+        }        
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();          
```

Comment:
In that case length will be `0`. But if a new change happens before first has finished this `this.__queuedChanges.push(newValue);` will happen after `let queuedChanges = this.__queuedChanges = [];` and before `if (!queuedChanges.length) {`. 

Easiest might be to try this out in an example, you can try out the code in skeleton with [this](https://github.com/aurelia/animator-css/issues/44) example, and add `console.log(queuedChanges.length)` [here](https://github.com/aurelia/templating-resources/blob/if-animation/src/if.js#L60), then trigger a few quick changes you will see a number get logged and decreased by one as each animation has finished until `0` and the function will return from the `if` statement.